allow Docker::Image to be refreshed

danielpanteleit opened the pull request:
I introduced another "exec" so that the "$image_install" can be triggered by external resources. The new "exec" checks if the image already exists and triggers the install if necessary. So the image is built if it does not exist or if it is refreshed externally.

With this change I can specify a "Dockerfile" resource that triggers the image on change.

garethr commented:
@daniellawrence nice idea. I noticed that this fails the unit tests. If you could update those to describe the behaviour that would be great.

This also sounds like something that would be useful to document in the README. Maybe add a quick example of how you're using it?

And thanks for the suggestion. Much appreciated.

danielpanteleit commented:
The failing tests were just about the no-op command "true" which seems puppet lint interprets as a quoted boolean value. I changed it to 'echo > /dev/null' and added a paragraph to the README.

danielpanteleit commented:
@garethr ready to be pulled
